Simbody/tests/TestSqrtMassMatrix.cpp

```diff
@@ -0,0 +1,280 @@
+                    pendulumBody,
+                        Transform(Rotation(randomAngle1, randomVecs[4]),
+                                  randomVecs[5]));
+
+    MobilizedBody::Slider
+        pendBody2a( pendBody2,
+                        Transform(Rotation(randomAngle1, randomVecs[6]),
+                                  randomVecs[7]),
+                    pendulumBody,
+                        Transform(Rotation(randomAngle2, randomVecs[8]),
+                                  randomVecs[9]));
+
+    MobilizedBody::Universal
+        pendBody2b( pendBody2a,
+                        Transform(Rotation(randomAngle1, randomVecs[8]),
+                                  randomVecs[7]),
+                    pendulumBody,
+                        Transform(Rotation(randomAngle2, randomVecs[6]),
+                                  randomVecs[5]));
+    MobilizedBody::Slider
+        pendBody2x( pendBody2b,
+                        Transform(Rotation(randomAngle1, randomVecs[6]),
+                                  randomVecs[7]),
+                    pendulumBody,
+                        Transform(Rotation(randomAngle2, randomVecs[8]),
+                                  randomVecs[9]));
+
+    MobilizedBody::Planar
+        pendBody4a( pendBody4,
+                        Transform(Rotation(randomAngle1, randomVecs[4]),
+                                  randomVecs[3]),
+                    pendulumBody,
+                        Transform(Rotation(randomAngle2, randomVecs[2]),
+                                  randomVecs[1]));
+
+
+    if (constrained) { // probably can't be satisfied, but doesn't matter
+        Constraint::Rod(pendBody4, pendBody2b, 1.); // holonomic
+        Constraint::ConstantSpeed
+            (pendBody2a, MobilizerUIndex(0), -3.); // nonholo
+        Constraint::ConstantAcceleration
+            (pendBody5, MobilizerUIndex(2), 0.01); // acc only
+        Constraint::Weld(pendBody4a, Test::randTransform(),
+                         pendBody4, Test::randTransform());
+    }
+}
+
+
+void testUnconstrainedSystem() {
+    MultibodySystem system;
+    MyForceImpl* frcp;
+    makeSystem(false, system, frcp);
+    const SimbodyMatterSubsystem& matter = system.getMatterSubsystem();
+
+    State state = system.realizeTopology();
+    const int nq = state.getNQ();
+    const int nu = state.getNU();
+    const int nb = matter.getNumBodies();
+
+    // Attainable accuracy drops with problem size.
+    const Real Slop = nu*SignificantReal;
+
+    system.realizeModel(state);
+    // Randomize state.
+    state.updQ() = Test::randVector(nq);
+    state.updU() = Test::randVector(nu);
+
+    Vector randVec = 100*Test::randVector(nu);
+
+    system.realize(state, Stage::Position);
+
+    Matrix MInv, SqrtMInv; 
+
+    // Compute SqrtMInv - calcSqrtMInv not yet implemented
+    SqrtMInv.resize(nu,nu);
+    const bool isContiguous = SqrtMInv(0).hasContiguousData();
+    Vector contig_col(isContiguous ? 0 : nu);
```

Comment:
You don't need the "contiguous data" stuff here -- your implementation already takes care of that internally.